apiserver/pkg/codeflare/appwrapper_builder.go

```diff
@@ -0,0 +1,167 @@
+	}
+	return rayCluster, nil
+}
+
+func (awb *defaultAppwrapperConverter) AppWrapperFromRayJob(rayJob *rayv1api.RayJob) (*mcadApi.AppWrapper, error) {
+	output, err := awb.rayJobToYAML(rayJob)
+	if err != nil {
+		return nil, fmt.Errorf("unable to serialize cluster to yaml: %w", err)
+	}
+	requests, limits := awb.calculateJobAggregateResources(rayJob)
+	jobAppWrapper := awb.makeAppWrapper(rayJob.Name, rayJob.Namespace, requests, limits, output)
+	return jobAppWrapper, nil
+}
+
+func (awb *defaultAppwrapperConverter) RayJobFromAppWrapper(jobAppWrapper *mcadApi.AppWrapper) (*rayv1api.RayJob, error) {
+	bytez := jobAppWrapper.Spec.AggrResources.GenericItems[0].GenericTemplate.Raw
+	codec := serializer.NewCodecFactory(awb.scheme, serializer.EnablePretty).LegacyCodec(rayv1api.SchemeGroupVersion)
+	rayJob := &rayv1api.RayJob{}
+	if err := runtime.DecodeInto(codec, bytez, rayJob); err != nil {
+		return nil, err
+	}
+	return rayJob, nil
+}
+
+func (awb *defaultAppwrapperConverter) calculateClusterAggregateResources(clusterHeadSpec *rayv1api.HeadGroupSpec, workerSpecs []rayv1api.WorkerGroupSpec) (corev1.ResourceList, corev1.ResourceList) {
+	var replicaCount int
+	requests := corev1.ResourceList{}
+	limits := corev1.ResourceList{}
+	awb.addContainersResources(clusterHeadSpec.Template.Spec.Containers, requests, limits)
+	awb.addContainersResources(clusterHeadSpec.Template.Spec.InitContainers, requests, limits)
+
+	for _, workerGroupSpec := range workerSpecs {
+		replicaCount += int(*workerGroupSpec.Replicas)
+		// calculate the resources for all the replica counts.
+		for i := 0; i < replicaCount; i++ {
+			// for each container in the worker group accumulate all the resources and the limits
+			awb.addContainersResources(workerGroupSpec.Template.Spec.Containers, requests, limits)
+			awb.addContainersResources(workerGroupSpec.Template.Spec.InitContainers, requests, limits)
+		}
+	}
+	return requests, limits
+}
+
+func (awb *defaultAppwrapperConverter) calculateJobAggregateResources(rayJob *rayv1api.RayJob) (corev1.ResourceList, corev1.ResourceList) {
+	requests := corev1.ResourceList{}
+	limits := corev1.ResourceList{}
+	if rayJob.Spec.RayClusterSpec != nil {
+		requests, limits = awb.calculateClusterAggregateResources(&rayJob.Spec.RayClusterSpec.HeadGroupSpec, rayJob.Spec.RayClusterSpec.WorkerGroupSpecs)
+	}
+	if rayJob.Spec.SubmitterPodTemplate == nil {
+		// Add default resources for the submitter pod
+		awb.addResourceLists(requests, corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("200m"),
+			corev1.ResourceMemory: resource.MustParse("256Mi"),
+		})
+		awb.addResourceLists(limits, corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("200m"),
+			corev1.ResourceMemory: resource.MustParse("256Mi"),
+		})
+	} else {
+		awb.addContainersResources(rayJob.Spec.SubmitterPodTemplate.Spec.Containers, requests, limits)
+		awb.addContainersResources(rayJob.Spec.SubmitterPodTemplate.Spec.InitContainers, requests, limits)
+	}
```

Comment:
The convoluted part here is that the job submitter does not have to have resources. It might also have incomplete resources, so its kind is finicky. Maybe drop job submitter? or just use default ones?